lti_consumer/lti_1p3/consumer.py

```diff
@@ -659,3 +661,228 @@ def enable_nrps(self, context_memberships_url):
+        """
+        Return claims specific to LTI Proctoring Services LtiStartProctoring LTI launch message,
+        to be injected into the LTI launch message.
+        """
+        proctoring_claims = self._get_base_claims()
+        proctoring_claims.update({
+            "https://purl.imsglobal.org/spec/lti/claim/message_type": "LtiStartProctoring",
+            "https://purl.imsglobal.org/spec/lti-ap/claim/start_assessment_url":
+                self.proctoring_data.get("start_assessment_url"),
+        })
+
+        return proctoring_claims
+
+    def get_end_assessment_claims(self):
+        """
+        Return claims specific to LTI Proctoring Services LtiEndAssessment LTI launch message,
+        to be injected into the LTI launch message.
+        """
+        proctoring_claims = self._get_base_claims()
+        proctoring_claims.update({
+            "https://purl.imsglobal.org/spec/lti/claim/message_type": "LtiEndAssessment",
+        })
+
+        return proctoring_claims
+
+    def generate_launch_request(
+        self,
+        preflight_response,
+        resource_link
+    ):
+        """
+        Build and return LTI launch message for proctoring.
+
+        This method overrides LtiConsumer1p3's method to include proctoring specific launch claims. It leverages
+        the set_extra_claim method to include these additional claims in the LTI launch message.
+        """
+        lti_message_hint = preflight_response.get("lti_message_hint")
+        proctoring_claims = None
+        if lti_message_hint == "LtiStartProctoring":
+            proctoring_claims = self.get_start_proctoring_claims()
+        elif lti_message_hint == "LtiEndAssessment":
+            proctoring_claims = self.get_end_assessment_claims()
+        else:
+            raise ValueError('lti_message_hint must be one of [LtiStartProctoring, LtiStartAssessment].')
+
+        self.set_extra_claim(proctoring_claims)
+
+        return super().generate_launch_request(preflight_response, resource_link)
+
+    def check_and_decode_token(self, token):
+        """
+        Once the Proctoring Tool is satisfied that the user has completed the necessary proctoring set up and that the
+        assessment will be proctored securely, it redirects the user to the Assessment Platform, directing the browser
+        to make a POST request containing a JWT and the session_data. This is the "Start Assessment" message.
+
+        This method validates the JWT signature and decodes the JWT. It also validates the claims in the JWT according
+        to the Proctoring Services specification.
+
+        It either returns a dictionary containing information required by the start_assessment, or it raises an
+        exception if any claims is missing or invalid.
+        """
+        # Decode token and check expiration.
+        proctoring_response = self.tool_jwt.validate_and_decode(token)
+
+        # TODO: We MUST perform other forms of validation here.
```

Comment:
I reread the specification, and now I cannot find where I read that we **MUST** do this validation. I swore I read it, but I cannot find it again. Based on that, I think the majority of this validation is not necessary. I may have misread a paragraph like the following as a requirement that we validate that the tool did its job. I don't think we _need_ to do that. That would mean that the `attempt_number` and `resource_link` would not longer be needed by the `start_assessment` view.  

> 4.3.1.6 Attempt number claim
The https://purl.imsglobal.org/spec/lti-ap/claim/attempt_number claim specifies the candidate's attempt number, as set by the Assessment Platform.  The Proctoring Tool MUST copy this value to the Start Assessment message unchanged.